src/System.Private.CoreLib/shared/Interop/Unix/System.Globalization.Native/Interop.Utils.cs

```diff
@@ -13,39 +15,44 @@ internal static partial class Interop
     /// increasing buffer until the size is big enough.
     /// </summary>
     internal static bool CallStringMethod<TArg1, TArg2, TArg3>(
-        Func<TArg1, TArg2, TArg3, StringBuilder, Interop.Globalization.ResultCode> interopCall,
+        SpanFunc<char, TArg1, TArg2, TArg3, Interop.Globalization.ResultCode> interopCall,
         TArg1 arg1,
         TArg2 arg2,
         TArg3 arg3,
         out string result)
     {
-        const int initialStringSize = 80;
-        const int maxDoubleAttempts = 5;
+        const int InitialStringSize = 256;
+        const int MaxDoubleAttempts = 3;
 
-        StringBuilder stringBuilder = StringBuilderCache.Acquire(initialStringSize);
+        Span<char> buffer = stackalloc char[InitialStringSize];
 
-        for (int i = 0; i < maxDoubleAttempts; i++)
+        for (int i = 0; i < MaxDoubleAttempts; i++)
         {
-            Interop.Globalization.ResultCode resultCode = interopCall(arg1, arg2, arg3, stringBuilder);
+            Interop.Globalization.ResultCode resultCode = interopCall(buffer, arg1, arg2, arg3);
 
             if (resultCode == Interop.Globalization.ResultCode.Success)
             {
-                result = StringBuilderCache.GetStringAndRelease(stringBuilder);
+                int length = buffer.IndexOf('\0');
+                Debug.Assert(length >= 0);
+                if (length >= 0)
```

Comment:
I do not think this defensive programming is needed.  We should always get zero-terminated string here because of this: https://github.com/dotnet/coreclr/blob/030a3ea9b8dbeae89c90d34441d4d9a1cf4a7de6/src/corefx/System.Globalization.Native/errors.h#L24

Reply:
Ok, will remove.